src/utils/ParseRouteUtils.js

```diff
@@ -676,14 +676,17 @@ function parseRoutes(
         const walkDepartureTime = convertMillisecondsToISOString(startTimeMs);
         const walkArrivalTime = convertMillisecondsToISOString(endTimeMs);
 
-        directions[0].startTime = walkDepartureTime;
-        directions[0].endTime = walkArrivalTime;
+        const walkDirections = {
+          ...directions[0],
+          startTime: walkDepartureTime,
+          endTime: walkArrivalTime,
```

Comment:
nifty use of the `...` syntax here, I love it

Reply:
creds to me :^)